packages/signals_core/lib/src/value/change_stack_signal.dart

```diff
@@ -0,0 +1,124 @@
+import 'dart:collection';
+
+import '../core/signals.dart';
+
+/// Change stack signal that can be used to call undo/redo on a value.
+///
+/// ```dart
+/// final s = ChangeStackSignal(0, limit: 5);
+/// s.value = 1;
+/// s.value = 2;
+/// s.value = 3;
+/// print(s.value); // 3
+/// s.undo();
+/// print(s.value); // 2
+/// s.redo();
+/// print(s.value); // 3
+/// ```
+class ChangeStackSignal<T> extends ValueSignal<T> {
```

Comment:
Is this a standard name elsewhere? I was expecting a name like ReplaySignal.

Reply:
Not standard per say, but what the data structure is. ReplaySignal is nice too. I could has a type alias for both.

I choose the name because of another package of mine:
https://pub.dev/packages/undo